### DISCUSSION TITLE
[3.5.0]: New License, Background colors and Statistics

### DISCUSSION THREAD

**Discussion by facontidavide** [facontidavide/PlotJuggler#686] [General]
# New License

I switched from LGPL to MPL-2.0 to make it easier to develop proprietary plugins.
Please remember that, if you modify an existing file in the code of PlotJuggler, you still need to release those modifications.

# Background colors

This is a very exciting new feature: you can highlight a portion of your plot using a background color based on the value of a timeseries. Potential applications:
- Link a "state" series to a set of colors, to show visually the state.
- Notify when a certain value exceeds a pre-defined threshold.

![2022-07-10_18-11](https://user-images.githubusercontent.com/2822888/178490207-00e98610-5de7-4ffc-b94c-eab6396f8829.png)

# Statistics dialog

Very simple floating dialog that display some basic statistics about the data in the visible region of a plot.
![2022-07-10_18-08](https://user-images.githubusercontent.com/2822888/178490594-1436544e-5972-4ebf-9ca9-7dcd99d47030.png)

<hr /><em>This discussion was created from the release <a href='https://github.com/facontidavide/PlotJuggler/releases/tag/3.5.0'>[3.5.0]: New License, Background colors and Statistics</a>.</em>

**Comment by AlexKlimaj** [comment]
Thanks for the Windows installer on this one!

**Comment by talldub** [comment]
I cant understand, how to use ColorMap function, may you make detailed description, please?

**Reply by facontidavide** [reply]
I should create a demo / video

**Reply by talldub** [reply]
It will be great
Or so for example: I put in colormap window function "more_than_10" with standard rule, like in your example. What to do next?
I feel, that I need to make custom series with this colormap function. But I tried to write this colormap function in custom series in different ways - it does not work at all

**Reply by facontidavide** [reply]
You need to drag and drop the series on the new icon on top-right corner

https://user-images.githubusercontent.com/2822888/179790166-2a20cdce-5e02-4c3b-bc70-2065d72bd5b6.mp4

**Reply by talldub** [reply]
Awesome! Thank you very much!

**Reply by talldub** [reply]
Last question :) Where does PlotJuggler store my ColorMap functions?

**Reply by facontidavide** [reply]
currently, it is saved with QSettings. 
In version 3.5.1, it will also be stored in the Layout files
